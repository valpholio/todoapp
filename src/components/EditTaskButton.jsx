import React from 'react';
import PropTypes from 'prop-types';

const EditTaskButton = ({
                            enterEditTaskMode = () => {
                            }, id
                        }) => {
    return (
            <button onClick={() => enterEditTaskMode(id)} className="btn btn-default btn-sm">
                <span className="glyphicon glyphicon-edit"></span> Edit
            </button>
    );
};

EditTaskButton.propTypes = {
    id: PropTypes.any.isRequired,
    enterEditTaskMode: PropTypes.func
};

export default EditTaskButton;