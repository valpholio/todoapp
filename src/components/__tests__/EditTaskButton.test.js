import React from 'react';
import renderer from 'react-test-renderer';
import EditTaskButton from '../EditTaskButton.jsx';

describe('EditTaskButton', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<EditTaskButton id='1'/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});