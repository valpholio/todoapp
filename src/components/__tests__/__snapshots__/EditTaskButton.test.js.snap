// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTaskButton snapshot renders 1`] = `
<button
  className="btn btn-default btn-sm"
  onClick={[Function]}
>
  <span
    className="glyphicon glyphicon-edit"
  />
   Edit
</button>
`;
